### PR TITLE
Cloudlift 2.1.0 Release

### PR DESCRIPTION
**Pull request by rokibulislaam**
Features:
 - Disable service alarms #155 
 - Cluster ALB #156 , #158, #159
 - Standalone executable support #153

Fixes:
- dnsmasq configuration issues on cluster template generator #154

### CODE REVIEW

File: cloudlift/deployment/service_template_generator.py [hunk 1]
```diff
@@ -700,19 +674,273 @@ def _add_alb(self, cd, service_name, config, launch_type):
...
+        https_listener_arns: list[str] = []
+
+        for listener in alb_listeners:
+            key = listener.get("Key")
+            if "HTTP" in key and "HTTPS" not in key:
+                if is_internal:
+                    # for internet facing services, we only need to create listener rules for HTTPS listener
+                    http_listener_arns.append(listener.get("Value"))
+            elif "HTTPS" in key:
+                https_listener_arns.append(listener.get("Value"))
```
Comment (shoan) [inline cloudlift/deployment/service_template_generator.py:838]: This can be refactored.

File: cloudlift/deployment/service_template_generator.py [hunk 2]
```diff
@@ -700,19 +674,273 @@ def _add_alb(self, cd, service_name, config, launch_type):
...
+            https_listener_rules.append(https_listener_rule)
+
+        if is_internal:
+            for rule in http_listener_rules:
+                self.template.add_resource(rule)
+            for rule in https_listener_rules:
+                self.template.add_resource(rule)
+        else:
+            for rule in https_listener_rules:
+                self.template.add_resource(rule)
```
Comment (shoan) [inline cloudlift/deployment/service_template_generator.py:868]: Refactor this

File: cloudlift/deployment/service_template_generator.py [hunk 3]
```diff
@@ -700,19 +674,273 @@ def _add_alb(self, cd, service_name, config, launch_type):
...
+
+        # get all the listener rules in the listener
+        elbv2_client = get_client_for('elbv2', self.env)
+        rules = elbv2_client.describe_rules(ListenerArn=listener_arn, PageSize=400)
+
+        # check if there's already a rule with same host-header condition
+        for rule in rules['Rules']:
+            for condition in rule['Conditions']:
+                if condition['Field'] == 'host-header' and hostname in condition['HostHeaderConfig']['Values']:
+                    return int(rule['Priority'])
```
Comment (shoan) [inline cloudlift/deployment/service_template_generator.py:883]: Need to also check that the priority of the rule returned is greater than `MIN_PRIORITY`

File: cloudlift/config/service_configuration.py [hunk 4]
```diff
@@ -317,29 +354,41 @@ def _validate_changes(self, configuration):
...
-                        str(".".join(list(validation_error.relative_path))))
+                path = []
+                for item in validation_error.relative_path:
+                    if isinstance(item, str):
+                        path.append(item)
+                    elif isinstance(item, int):
+                        path.append(str(item))
+                    else:
+                        path.append(str(item))
+                path_string = ".".join(path)
```
Comment (shoan) [inline cloudlift/config/service_configuration.py:365]: Refactor

**Review by shoan — CHANGES_REQUESTED** [review]

File: cloudlift/deployment/service_template_generator.py (hunk 1 above)
Comment (rokibulislaam) [reply cloudlift/deployment/service_template_generator.py]: they have been refactored.

File: cloudlift/deployment/service_template_generator.py (hunk 3 above)
Comment (rokibulislaam) [reply cloudlift/deployment/service_template_generator.py]: changed the behavior, reduced MAX_PRIORITY from  49999 to 49899, room for 100 more custom rules after the priorities generated by the code.

File: cloudlift/config/service_configuration.py (hunk 4 above)
Comment (rokibulislaam) [reply cloudlift/config/service_configuration.py]: Refactored it.

File: cloudlift/deployment/service_template_generator.py (hunk 2 above)
Comment (rokibulislaam) [reply cloudlift/deployment/service_template_generator.py]: They have been refactored.

**Review by shoan — APPROVED** [review]

**Pull request merged**